test/flink_test.go

```diff
@@ -120,3 +152,214 @@ func (s *CLITestSuite) TestFlink_Autocomplete() {
+		require.NoError(t, err)
+		defer pipe.Close()
+		stdoutScanner := bufio.NewScanner(pipe)
+
+		// Start command
+		err = cmd.Start()
+		require.NoError(t, err)
+		go killCommandIfTimeoutExceeded(flinkShellTest.timeout, cmd)
+
+		output := &strings.Builder{}
+		waitForFlinkShellToBeReady(stdoutScanner, output)
+
+		// Execute commands
+		commands, err := getCommandsFromFixture(filepath.Join(dir, flinkShellFixtureInputFolder, flinkShellTest.inputFile))
+		require.NoError(t, err)
+		err = executeCommands(stdin, commands, stdoutScanner, output)
+		require.NoError(t, err)
+
+		// Wait for flink shell to exit
+		err = cmd.Wait()
+		require.NoError(t, err)
+
+		// Compare to golden file
+		snapshotConfig := cupaloy.New(
+			cupaloy.SnapshotSubdirectory(filepath.Join(dir, flinkShellFixtureOutputFolder)),
+			// Update snapshot if update flag was set
+			cupaloy.ShouldUpdate(func() bool {
+				return *update
+			}),
+		)
+		assert.NoError(t, snapshotConfig.SnapshotWithName(flinkShellTest.goldenFile, output.String()),
+			fmt.Sprintf("full output was %s", output.String()))
+	})
+}
+
+func (s *CLITestSuite) login(t *testing.T) {
+	loginString := fmt.Sprintf("login --url %s", s.TestBackend.GetCloudUrl())
+	env := []string{pauth.ConfluentCloudEmail + "=fake@user.com", pauth.ConfluentCloudPassword + "=pass1"}
+	for _, e := range env {
+		keyVal := strings.Split(e, "=")
+		os.Setenv(keyVal[0], keyVal[1])
+	}
+
+	defer func() {
+		for _, e := range env {
+			keyVal := strings.Split(e, "=")
+			os.Unsetenv(keyVal[0])
+		}
+	}()
+
+	if output := runCommand(t, testBin, env, loginString, 0, ""); *debug {
+		fmt.Println(output)
+	}
+}
+
+func cleanupInputFile(file *os.File) {
+	err := file.Close()
+	if err != nil {
+		fmt.Printf("failed to close file: %v\n", err)
+	}
+	err = os.Remove(file.Name())
+	if err != nil {
+		fmt.Printf("failed to remove file: %v\n", err)
+	}
+}
+
+func setEnvVars(inputFileName string) ([]string, error) {
+	var envVars []string
+
+	// Set the go-prompt file input env var, so go-prompt uses this file as the input stream
+	if err := os.Setenv(prompt.EnvVarInputFile, inputFileName); err != nil {
+		return envVars, err
+	}
+	envVars = append(envVars, prompt.EnvVarInputFile)
+
+	// Fake the timezone, to ensure CI and local run with the same default timezone
+	if err := os.Setenv(timezoneEnvVar, "Europe/London"); err != nil {
+		return envVars, err
+	}
+	envVars = append(envVars, timezoneEnvVar)
+
+	return envVars, nil
+}
+
+func cleanupEnvVars(envVars []string) {
+	for _, envVar := range envVars {
+		os.Unsetenv(envVar)
+	}
+}
+
+func killCommandIfTimeoutExceeded(timeout time.Duration, cmd *exec.Cmd) {
+	if timeout == 0 {
+		timeout = 10 * time.Second
+	}
+	time.Sleep(timeout)
+	if err := cmd.Process.Kill(); err != nil {
+		fmt.Printf("failed to kill command %v/n", err)
+	}
+}
+
+func waitForFlinkShellToBeReady(scanner *bufio.Scanner, output *strings.Builder) {
+	for scanner.Scan() {
+		line := scanner.Text()
+		output.WriteString(line + "\n")
```

Comment:
do we have to execute this for every scan before checking if strings.Contains(line, "[Ctrl-Q]..?

Reply:
Ah, I understand! You do this to print whatever we have as welcome message/headers when starting the CLI.